### ISSUE TITLE
Can't open display :0

### ISSUE THREAD

**Issue by mishoboss** [Salamek/chromium-kiosk#81]
Hi, after a year with OrangePi 4 I upgraded to Orange Pi 5 and the newest version of chromium-kiosk. However experience is not a smooth one this time either. I have hooked the same display via type-c and it properly displays picture. However it seems chromium-kiosk doesn't see it. Any suggestions how to debug it?

<img width="953" alt="image" src="https://github.com/Salamek/chromium-kiosk/assets/1629929/97f301ce-ed24-4534-9cca-c09285c52978">

**Comment by Salamek** [comment]
chromium-kiosk should be run under chromium-kiosk user under witch X11 should be running...

**Comment by Salamek** [comment]
@mishoboss any news on this? Did you try running it under correct user?

**Comment by mishoboss** [comment]
I just installed it from scratch, swicthed to `chromium-kiosk` user, but same result:

<img width="988" alt="image" src="https://github.com/Salamek/chromium-kiosk/assets/1629929/03e4d6cc-8db3-4ec0-bff8-521ae0f4fc53">


BTW, how can I know the `chromium-kiosk` user password?

**Comment by mishoboss** [comment]
I switch to root during installation for adding the GPG key and the repository, cause sudo is not enough for some reason. That shouldn't affect anything, but just to mention it. The actual chromium-kiosk installation I do using the non-root user provided by orangepi.

**Comment by Salamek** [comment]
chromium-kiosk have disabled password, if you want to set one use `passwd chromium-kiosk`

Back to the issue, does screen (X11 server) try to start on boot or you end up in console without any attempt to start X11?
It is possible that startup script is not started correctly for some reason.
If that is a case try to login as chromium-kiosk and run command `startx`, if that works we have issue with startup script

**Comment by mishoboss** [comment]
Switched to chromium-kiosk user and executed startx:

<img width="1316" alt="image" src="https://github.com/Salamek/chromium-kiosk/assets/1629929/1f159683-68b0-47ae-9d3b-ea394b587b11">

**Comment by mishoboss** [comment]
I added the `chromium-kiosk` to sudoers and tried with sudo:

<img width="1310" alt="image" src="https://github.com/Salamek/chromium-kiosk/assets/1629929/3bd97efc-f55b-4ec6-bbc5-17bca667415c">

**Comment by Salamek** [comment]
Hmm maybe try login instead of su?

**Comment by mishoboss** [comment]
Absolutely same result. :(

**Comment by Salamek** [comment]
Try to put chromium-kiosk user to tty group, relogin as chromium-kiosk user and try again?
```
usermod -a -G tty chromium-kiosk
```
Also what OS are you running? Can you provide link to installation image? And what steps did you do to install chromium kiosk? (Standard repository install?) Ill try to run it in qemu or something to debug this i think...

**Comment by mishoboss** [comment]
1. I added the user to tty group, relogined, but same result.
2. Debain 12 Bookworm for OrangePi 5. I tried both with kernel 5.10.110 and 5.10.160
3. https://drive.google.com/drive/folders/1F2uc8v_EQnvsNrevDihwoymOJlFgM-dZ

**Comment by mishoboss** [comment]
Also, I strictly follow the installation steps in the docs. The only things I change is either I switch to root to add GPG key and repo, or I edit those two lines like this:

`sudo wget -O- https://repository.salamek.cz/deb/salamek.gpg | sudo tee /usr/share/keyrings/salamek-archive-keyring.gpg > /dev/null`

`sudo echo "deb     [signed-by=/usr/share/keyrings/salamek-archive-keyring.gpg] https://repository.salamek.cz/deb/pub all main" | sudo tee /etc/apt/sources.list.d/salamek.cz.list`

**Comment by Salamek** [comment]
That URL you have provided is official source?

**Comment by mishoboss** [comment]
Yes, official source. Chinese...

**Comment by Salamek** [comment]
It will take me some time to test, in the mean time (if you have time), what about to try Armbian instead? (that is what i use on my OrangePi PC+s)

https://www.armbian.com/orangepi-5/  Armbian 24.2.1 Bookworm CLI

**Comment by mishoboss** [comment]
Mainline kernel doesn't push image to type-c and hdmi. I'll try it though.

**Comment by mishoboss** [comment]
Armbian works like a charm.

**Comment by Salamek** [comment]
Chromium kiosk works on Armbian install OK?

**Comment by mishoboss** [comment]
Yes, it starts and displays fine the browser. However I don't have a working touchscreen. I remember back then a year ago we did some magic tricks because I had the same issue.

**Comment by Salamek** [comment]
It was this issue https://github.com/Salamek/chromium-kiosk/issues/72 i guess, but that got fixed...

login as chromium-kiosk user via SSH and run `chromium-kiosk info`

**Comment by mishoboss** [comment]
I guess `chromium-kiosk system_info`:

<img width="556" alt="image" src="https://github.com/Salamek/chromium-kiosk/assets/1629929/b97af4e1-6df5-49ef-ac0c-1e0ec328ff3c">

**Comment by Salamek** [comment]
Oh yes `system_info` sorry, hmm no touchscreen device detected, how is your touchscreen connected?

Run 
```
DISPLAY=:0 xinput -list
```

under chromium-kiosk via SSH

**Comment by mishoboss** [comment]
Seems it doesn't see it:

<img width="650" alt="image" src="https://github.com/Salamek/chromium-kiosk/assets/1629929/6c252c50-0d9e-496e-99e4-ee275a7fbea6">

**Comment by mishoboss** [comment]
Touchscreen is connected via USB Type-C. Same display works fine when hooked to MacBook Pro, so I'm sure touchscreen works fine on Type-C. Issue is something with the distro.

**Comment by Salamek** [comment]
Hmm USB,  do you see it in `lsusb` (under root)?, maybe check `dmesg` for system messages of missing firmware or something

**Comment by mishoboss** [comment]
<img width="548" alt="image" src="https://github.com/Salamek/chromium-kiosk/assets/1629929/28f4e56e-8fcb-4eb4-9a29-47eae8cbf9c7">

In dmesg logs I found this:

```
[    7.183238] Error: Driver 'Goodix-TS' is already registered, aborting...
[    7.183268] usbcore: registered new interface driver usbtouchscreen
```

**Comment by mishoboss** [comment]
```
[    7.203062] Goodix-TS 2-0014: Looking up AVDD28-supply from device tree
[    7.203070] Goodix-TS 2-0014: Looking up AVDD28-supply property in node /i2c@feaa0000/touchscreen@14 failed
[    7.203081] Goodix-TS 2-0014: supply AVDD28 not found, using dummy regulator
[    7.203130] Goodix-TS 2-0014: Looking up VDDIO-supply from device tree
[    7.203137] Goodix-TS 2-0014: Looking up VDDIO-supply property in node /i2c@feaa0000/touchscreen@14 failed
[    7.203144] Goodix-TS 2-0014: supply VDDIO not found, using dummy regulator
[    7.296449] Goodix-TS 2-0014: i2c test failed attempt 1: -6
[    7.323110] Goodix-TS 2-0014: i2c test failed attempt 2: -6
[    7.349632] Goodix-TS 2-0014: I2C communication failure: -6
```

**Comment by Salamek** [comment]
I don't think that is it, this is i2c driver not USB and i have found same error message in dmesg posted by some Orange-pi-5 user running armbian without touchscreen connected... nothing else there?

**Comment by mishoboss** [comment]
These are the only touch related lines:

```
[    9.839775] usb 7-1.2: new full-speed USB device number 3 using xhci-hcd
[   10.032611] usb 7-1.2: New USB device found, idVendor=0416, idProduct=c168, bcdDevice= 0.00
[   10.032624] usb 7-1.2: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[   10.032632] usb 7-1.2: Product: MTouch
[   10.032640] usb 7-1.2: Manufacturer: TSTP
[   10.032648] usb 7-1.2: SerialNumber: CMTP_1.0
```


```
[    7.183238] Error: Driver 'Goodix-TS' is already registered, aborting...
[    7.183268] usbcore: registered new interface driver usbtouchscreen
```

**Comment by Salamek** [comment]
Hmm is `xserver-xorg-input-evdev` installed? (it should be but who knows)

**Comment by mishoboss** [comment]
Actually it wasn't installed. I installed, restarted and tested, but no luck again. Also, I run `chromium-kiosk system_info` and `DISPLAY=:0 xinput -list`, but same results as before.

**Comment by Salamek** [comment]
Can you run `lsusb -t` ?

**Comment by mishoboss** [comment]
<img width="625" alt="image" src="https://github.com/Salamek/chromium-kiosk/assets/1629929/01e400ea-67cb-468e-948c-48306ed5216b">

**Comment by Salamek** [comment]
Ok this is weird, in dmesg we can see that it found USB trouchscreen `New USB device found, idVendor=0416, idProduct=c168,` (https://linux-hardware.org/?id=usb:0416-c168) but it is not listed in `lsusb` actualy your usb tree is weirdly empty... can you connect a mouse to it for example and try that USB subsystem is working correctly (that mouse works) sounds like this issue:
https://forum.radxa.com/t/usb-touch-screen-kills-usb/13538/3

Someone is suggesting to disable OTG usb to fix this... some that replacing power adapter helped... but from i'm finding it is a small miracle it worked for you the first time :-)

**Comment by mishoboss** [comment]
Keyboard works like a charm and is listed properly when hooked. It's apparently something with the display itself. I tried with two other power supplies, and also I use type-c from OrangePi to power the display, so grounding issue should be eliminated that way.

No idea how this worked before, but I had a very different setup - another board (OrangePi 4), another power supply, another connection setup - HDMI and USB type A to type A for touch. I had issues back then as well, but eventually they were overcomed somehow.

**Comment by Salamek** [comment]
Well this is where my support in this matter ~ends, this is on 80% HW issue (20% driver issue) i can't solve without getting my hands on same HW you have (i actually have homelab with equipment to debug that lol).
You may try to experiment with some versions of distros with DE for your board to check if touch works and if it does on some, bisect the shit out of it (kernel versions, loaded kernel modules, applied patches by distro team, etc) Or if you find some distro with desktop that works and don't want to hunt what the issue is, just run chromium-kiosk/qiosk on top of working full DE distro by removing chromium-kiosk getty@tty1.service.d/override.conf and adding chromium-kiosk run in "run after start" of used distro... (yes it is bit hacky and resource wasteful but should work ~OK)

**Comment by mishoboss** [comment]
Thanks. It's probably a hardware issue indeed, so yes, I can't expect your support on this. But HUGE THANKS for the whole support so far. You're an awesome engineer.

I have the same experience with x86 SBCs - there are some low power Celeron options out there with passive cooling, so it's not that bad, just more expensive and a bit bigger (which is an issue when you put it in a wall).

Thanks one more time!

**Comment by Salamek** [comment]
You are welcome , good luck!
